### PR TITLE
Add operator lexing

### PR DESCRIPTION
**Pull request by adamhutchings**

### CODE REVIEW

**Comment by adamhutchings** [comment]
Do not want to merge yet until token type identification is fixed.

**Comment by TheCrystalKeeper** [comment]
Seems like it doesn't break anything, but I'm convinced that the code
```c
        ungetc(c, l->fp);
        t->contents[pos - 1] = '\0';
        t->type = ttype_from_string(t->contents);
        t->length = pos;
        return 0;
```

doesn't need to be duplicated inside of `lex()`

**Comment by adamhutchings** [comment]
I could `#define` a macro to do that, or `goto` an end block which would be called `finalize_token` or something similar. What do you think?

**Comment by TheCrystalKeeper** [comment]
Try adding a return 1 to those places where you want to void the values of t. Then inside of lex() if it is determined to be 1, set t appropriately. I think that makes it just clearer.

File: src/lexer/lex.c [hunk 1]
```diff
@@ -143,6 +201,22 @@ int real_lex(Lexer *l, Token *t) {
...
 
+    // Lex an operator. We do this by lexing characters from the buffer until
+    // the resulting string is no longer an operator, then we cut our losses and
+    // return./
+    if (starts_operator(init)) {
+        while (valid_operator_sequence(t->contents)) {
+            t->contents[pos++] = (c = getc(l->fp));
+        }
+        // We've ended!
+        // Can we reduce this code duplication from above in a smart way?
```
Comment (JakeRoggenbuck) [inline src/lexer/lex.c:212]: A few lines is okay imo

**Review by JakeRoggenbuck — APPROVED** [review]

**Pull request merged**